R/sim_linelist.R

```diff
@@ -0,0 +1,217 @@
+#' )
+sim_linelist <- function(R,
+                         serial_interval,
+                         onset_to_hosp,
+                         onset_to_death,
+                         hosp_rate = 0.2,
+                         hosp_death_rate = 0.5,
+                         non_hosp_death_rate = 0.05,
+                         outbreak_start_date = as.Date("2023-01-01"),
+                         add_names = TRUE,
+                         add_ct = FALSE,
+                         min_chain_size = 10,
+                         population_age = c(1, 90),
+                         case_type_probs = c(
+                           suspected = 0.2,
+                           probable = 0.3,
+                           confirmed = 0.5
+                         ),
+                         config = create_config(),
+                         ...) {
+  chkDots(...)
+
+  serial_interval <- as.function(serial_interval, func_type = "generate")
+  onset_to_hosp <- as.function(onset_to_hosp, func_type = "generate")
+  onset_to_death <- as.function(onset_to_death, func_type = "generate")
+
+  .check_sim_input(
+    sim_type = "linelist",
+    R = R,
+    serial_interval = serial_interval,
+    outbreak_start_date = outbreak_start_date,
+    min_chain_size = min_chain_size,
+    onset_to_hosp = onset_to_hosp,
+    onset_to_death = onset_to_death,
+    add_names = add_names,
+    add_ct = add_ct,
+    case_type_probs = case_type_probs,
+    hosp_rate = hosp_rate,
+    hosp_death_rate = hosp_death_rate,
+    non_hosp_death_rate = non_hosp_death_rate,
+    population_age = population_age
+  )
+
+  if (is.data.frame(population_age)) {
+    population_age <- .check_age_df(population_age)
+    age_range <- c(
+      min(population_age[, "min_age"]),
+      max(population_age[, "max_age"])
+    )
+  } else {
+    age_range <- population_age
+  }
+  if (is.data.frame(hosp_rate)) {
+    hosp_rate <- .check_rate_df(
+      hosp_rate,
+      age_range = age_range
+    )
+  }
+  if (is.data.frame(hosp_death_rate)) {
+    hosp_death_rate <- .check_rate_df(
+      hosp_death_rate,
+      age_range = age_range
+    )
+  }
+  if (is.data.frame(non_hosp_death_rate)) {
+    non_hosp_death_rate <- .check_rate_df(
+      non_hosp_death_rate,
+      age_range = age_range
+    )
+  }
+
+  chain <- .sim_bp_linelist(
+    R = R,
+    serial_interval = serial_interval,
+    outbreak_start_date = outbreak_start_date,
+    min_chain_size = min_chain_size,
+    population_age = population_age,
+    config = config
+  )
+
+  linelist <- .sim_clinical_linelist(
+    chain = chain,
+    onset_to_hosp = onset_to_hosp,
+    onset_to_death = onset_to_death,
+    hosp_rate = hosp_rate,
+    hosp_death_rate = hosp_death_rate,
+    non_hosp_death_rate = non_hosp_death_rate,
+    outbreak_start_date = outbreak_start_date,
+    add_names = add_names,
+    add_ct = add_ct,
+    case_type_probs = case_type_probs,
+    config = config
+  )
+
+  chain <- linelist$chain[, linelist$cols]
```

Comment:
This would be fine for `sim_linelist()` but I think this would cause issues in `sim_outbreak()`, where the `$chain` output of `.sim_clinical_linelist()` is used by `.sim_contacts_tbl()` before subsetting with `$col`.